includes/sanitizers/class-amp-gallery-block-sanitizer.php

```diff
@@ -69,104 +71,60 @@ public function sanitize() {
 				]
 			)
 		);
-		$query       = $this->dom->xpath->query( $expr );
-
-		$nodes = [];
-		foreach ( $query as $node ) {
-			$nodes[] = $node;
-		}
+		$nodes       = $this->dom->xpath->query( $expr );
 
 		foreach ( $nodes as $node ) {
+			/** @var DOMElement $node */
+
 			// In WordPress 5.3, the Gallery block's <ul> is wrapped in a <figure class="wp-block-gallery">, so look for that node also.
 			$gallery_node = isset( $node->parentNode ) && AMP_DOM_Utils::has_class( $node->parentNode, self::$class ) ? $node->parentNode : $node;
 			$attributes   = AMP_DOM_Utils::get_node_attributes_as_assoc_array( $gallery_node );
 
-			$is_amp_lightbox = isset( $attributes['data-amp-lightbox'] ) && true === filter_var( $attributes['data-amp-lightbox'], FILTER_VALIDATE_BOOLEAN );
-			$is_amp_carousel = (
-				! empty( $this->args['carousel_required'] )
-				||
-				filter_var( $node->getAttribute( 'data-amp-carousel' ), FILTER_VALIDATE_BOOLEAN )
-				||
-				filter_var( $node->parentNode->getAttribute( 'data-amp-carousel' ), FILTER_VALIDATE_BOOLEAN )
-			);
-
-			// We are only looking for <ul> elements which have amp-carousel / amp-lightbox true.
-			if ( ! $is_amp_carousel && ! $is_amp_lightbox ) {
-				continue;
-			}
-
-			// If lightbox is set, we should add lightbox feature to the gallery images.
-			if ( $is_amp_lightbox ) {
-				$this->add_lightbox_attributes_to_image_nodes( $node );
-			}
+			$gallery_node->removeAttribute( 'data-amp-carousel' );
+			$gallery_node->removeAttribute( 'data-amp-lightbox' );
```

Comment:
If I comment out these lines, then the `amp-carousel` renders as expected for some reason.

Reply:
OK that's weird, as far as I know those attributes should not be interfering with how the `amp-carousel` is to be rendered.

Reply:
Actually, it seems we need those attributes as we are actually using it for styling:

https://github.com/ampproject/amp-wp/blob/e25251f73e59756d9200d18ba748ea62c24c9b78/assets/css/src/amp-default.css#L76-L79

Reply:
Oh right, yea that makes sense. OK I'll revert that change.

I'm also chasing a potential bug with the migration of `data-amp-layout`. Given the following markup:

```html
<!-- wp:image {"id":570,"sizeSlug":"large","ampLayout":"responsive"} -->
<figure class="wp-block-image size-large" data-amp-layout="responsive"><img src="https://wordpressdev.lndo.site/content/uploads/2020/04/axs14htjmbo.jpeg" alt="man in gray crew neck shirt wearing black framed eyeglasses" class="wp-image-570"/><figcaption>responsive layout</figcaption></figure>
<!-- /wp:image -->
```

Whenever the block is updated in the editor and the post is saved, a validation error occurs saying `data-amp-layout` should not be `true`.

Reply:
Bug related to the `data-amp-layout` attribute squashed in https://github.com/ampproject/amp-wp/pull/4775/commits/f7be98b50582bd6f40d2ab40be8929812978609c. There's one edge case with the `amp-carousel` I'm working on to resolve and that should _hopefully_ be it.

Reply:
Not quite as this resulted in `data-amp-carousel="1"` which then didn't match the selector in `amp-default.css`. Follow up in da088e0c1.